src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,333 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useConnect, useConnectors } from 'wagmi';
+import { coinbaseWallet } from 'wagmi/connectors';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { coinbaseWalletSvg } from '../../internal/svg/coinbaseWalletSvg';
+import { defaultAvatarSVG } from '../../internal/svg/defaultAvatarSVG';
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  className?: string;
+  onError?: (error: Error) => void;
+};
+
+export function WalletModal({
+  isOpen,
+  onClose,
+  className,
+  onError,
+}: WalletModalProps) {
+  const modalRef = useRef<HTMLDivElement>(null);
+  const [shouldRender, setShouldRender] = useState(isOpen);
```

Comment:
why do we need `shouldRender` if we have `isOpen`? is this just to handle animation delays?